crates/wash-lib/src/run/mod.rs

```diff
@@ -0,0 +1,95 @@
+use crate::registry::get_oci_artifact;
+use crate::run::ctx::Ctx;
+use anyhow::{Context, Result};
+use wasmtime::component::Component;
+use wasmtime::{Config, Engine};
+use wasmtime_wasi::{WasiCtx, WasiView};
+
+pub use crate::run::ctx::CtxBuilder;
+
+pub use wasmtime_wasi::{DirPerms, FilePerms};
+
+pub mod ctx;
+
+/// Runs workloads capable of leveraging the local host it runs on.
+///
+/// Access to the host's resources can be strictly controlled.
+/// Follows the principle of least privilege, which means all is denied by
+/// default.
+pub struct LocalRuntime {
+    engine: Engine,
+}
+
+impl LocalRuntime {
+    pub fn new() -> Result<Self> {
+        let mut config = Config::new();
+
+        // Try to read settings from default location
+        if let Err(e) = config.cache_config_load_default() {
+            tracing::warn!(err = ?e, "Failed to load wasm cache");
+        }
+
+        config.wasm_component_model(true);
+        config.async_support(true);
+
+        let engine = Engine::new(&config)?;
+        anyhow::Ok(Self { engine })
+    }
+
+    pub async fn run(&self, context: Ctx) -> Result<()> {
+        let artifact = get_oci_artifact(
+            context.reference.clone(),
+            context.oci_cache_file.clone(),
+            context.oci_pull_options.clone(),
+        )
+        .await
+        .context("failed to pull the component")?;
```

Comment:
❓ Is there a plan to harmonize how we resolve artifacts from strings?

I saw @thomastaylor312 re-implemented that in wash plugin so I am not sure which approach I should take.

Reply:
I really like using the URI approach like I did in wash plugins so it can be really clear (e.g. `file://` and `oci://`)

Reply:
I am 100% aligned with you, what we lose in "simplicity" (because we now need to add an URI), we gain in determinism: the resolution of your artifact is now more "explicit" there is less "magic" involved.